react/features/participants-pane/components/web/VisitorsList.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useCallback } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch, useSelector } from 'react-redux';
+import { makeStyles } from 'tss-react/mui';
+
+import { IReduxState } from '../../../app/types';
+import { withPixelLineHeight } from '../../../base/styles/functions.web';
+import { admitMultiple } from '../../../visitors/actions';
+import { getPromotionRequests } from '../../../visitors/functions';
+
+import { VisitorsItem } from './VisitorsItem';
+
+const useStyles = makeStyles()(theme => {
+    return {
+        container: {
+            margin: `${theme.spacing(3)} 0`
+        },
+        headingW: {
+            color: theme.palette.warning02
+        },
+        drawerActions: {
+            listStyleType: 'none',
+            margin: 0,
+            padding: 0
+        },
+        drawerItem: {
+            alignItems: 'center',
+            color: theme.palette.text01,
+            display: 'flex',
+            padding: '12px 16px',
+            ...withPixelLineHeight(theme.typography.bodyShortRegularLarge),
+
+            '&:first-child': {
+                marginTop: '15px'
+            },
+
+            '&:hover': {
+                cursor: 'pointer',
+                background: theme.palette.action02
+            }
+        },
+        icon: {
+            marginRight: 16
+        },
+        headingContainer: {
+            alignItems: 'center',
+            display: 'flex',
+            justifyContent: 'space-between'
+        },
+        heading: {
+            ...withPixelLineHeight(theme.typography.bodyShortBold),
+            color: theme.palette.text02
+        },
+        link: {
+            ...withPixelLineHeight(theme.typography.labelBold),
+            color: theme.palette.link01,
+            cursor: 'pointer'
+        }
+    };
+});
+
+/**
+ * Component used to display a list of visitors waiting for approval to join the main meeting.
+ *
+ * @returns {ReactNode}
+ */
+export default function Visitors() {
```

Comment:
```suggestion
// Memoize the component in order to avoid rerender on drawer open/close.
export default React.memo<IProps>(function VisitorsList() {
```